ansible/roles/ci/code_style_check/license/files/copyright_checks.py

```diff
@@ -0,0 +1,497 @@
+#!/usr/bin/env python3
+
+# Copyright 2023 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import argparse
+import datetime
+import logging
+import os
+import re
+import subprocess
+import sys
+
+# We want to use f-strings for logging. Very minor performance hit, better consistency.
+# pylint: disable=logging-fstring-interpolation
+
+
+class CopyrightChecker:
```

Comment:
I'm a bit confused. I thought this functionality was already part of our CI infrastructure.
Is this a duplication or a copy of scripts that we have elsewhere?